src/hooks/useDeployment.ts

```diff
@@ -0,0 +1,110 @@
+import { useState, useCallback } from "react";
+
+import { MaciService } from "~/services/maci";
+import { useEthersSigner } from "./useEthersSigner";
+import { eas } from "~/config";
+
+interface IDeployArgs {
+  amount: number;
+}
+
+export function useDeployment() {
+  /**
+   * deployStatus:
+   * 0. default
+   * 1. VoiceCreditProxy
+   * 2. Gatekeeper
+   * 3. Verifier
+   * 4. TopupCredit
+   * 5. Poseidon contracts
+   * 6. PollFactory
+   * 7. MessageProcessorFactory
+   * 8. TallyFactory
+   * 9. MACI contract
+   * 10. VkRegistry
+   */
+  const [deployStatus, setDeployStatus] = useState<number>(0);
+  const signer = useEthersSigner();
+
+  const maci = new MaciService(signer);
+
+  const deploy = useCallback(
+    async ({ amount }: IDeployArgs) => {
+      const attester = await signer?.getAddress();
+      if (!signer || !attester) console.error("no signer");
+
+      // 1. deploy voice credit proxy contract
+      setDeployStatus(1);
+      await maci.deployInitialVoiceCreditProxy({ amount });
+
+      // 2. deploy EASGatekeeper contract
+      setDeployStatus(2);
+      await maci.deployGatekeeper({
+        easAddress: eas.contracts.eas,
+        encodedSchema: eas.schemas.approval,
+        attester,
+      });
+
+      // 3. deploy verifier contract
+      setDeployStatus(3);
+      await maci.deployVerifier();
+
+      // 4. deploy topup credit contract
+      setDeployStatus(4);
+      await maci.deployTopupCredit();
+
+      // 5. deploy Poseidon contracts
+      setDeployStatus(5);
+      await maci.deployPoseidon();
+
+      // 6. deploy PollFactory contract
+      setDeployStatus(6);
+      await maci.deployPollFactory();
+
+      // 7. deploy MessageProcessorFactory contract
+      setDeployStatus(7);
+      await maci.deployMessageProcessorFactory();
+
+      // 8. deploy TallyFactory contract
+      setDeployStatus(8);
+      await maci.deployTallyFactory();
+
+      // 9. deploy MACI contract
+      setDeployStatus(9);
+      await maci.deployMaci({ stateTreeDepth: 10 });
+
+      // 10. deploy VkRegistry contract
+      setDeployStatus(10);
+      const processMessagesZkeyQv = await maci.extractVk(
+        "/zkeys/ProcessMessages_10-2-1-2_test/ProcessMessages_10-2-1-2_test.0.zkey",
+      );
+      const processMessagesZkeyNonQv = await maci.extractVk(
+        "/zkeys/ProcessMessagesNonQv_10-2-1-2_test/ProcessMessagesNonQv_10-2-1-2_test.0.zkey",
+      );
+      const tallyVotesZkeyQv = await maci.extractVk(
+        "/zkeys/TallyVotes_10-1-2_test/TallyVotes_10-1-2_test.0.zkey",
+      );
+      const tallyVotesZkeyNonQv = await maci.extractVk(
+        "/zkeys/TallyVotesNonQv_10-1-2_test/TallyVotesNonQv_10-1-2_test.0.zkey",
+      );
+
+      await maci.deployVkRegistry({
+        stateTreeDepth: 10,
+        intStateTreeDepth: 1,
+        messageTreeDepth: 2,
+        messageBatchDepth: 1,
+        voteOptionTreeDepth: 2,
```

Comment:
Use constants for these values

Reply:
do you think they could be changed by the coordinator if they want?
if no, then would make them constants;
if yes, then they could be environmental parameters or input arguments.

Reply:
It will depend on zkey configuration but we can control these files some I'd move these values to constants